src/lib/comms/sol-coap.c

```diff
@@ -763,6 +731,8 @@ prepare_buffer(struct outgoing *outgoing, struct sol_buffer *buffer)
         payload->buf.used - old_offset);
     SOL_INT_CHECK_GOTO(r, < 0, error);
 
+    buffer->used = new_size;
```

Comment:
this shouldn't be needed as both appends will increment the size. Just be aware that the fixed-capacity doesn't prevent trailing null-bytes. I'm not sure append_bytes are appending them, but they should. To prevent you can use the `NO_NUL_BYTE` flag.